tpgtools/main_helpers.go

```diff
@@ -104,3 +107,25 @@ func renderDefault(t Type, val string) (string, error) {
 	}
 	return "", fmt.Errorf("Failed to find default format for type: %v", t)
 }
+
+// ConvertSampleJSONToDCLResource unmarshals json to a DCL resource specified by th resource type
+// TODO: This should be generated in serialization template
```

Comment:
What is the plan on this? Are we planning to include this in the DCL?

Reply:
I would feel weird about adding a feature that needs a switch statement for each resource we support

Reply:
The main problem is that we need to map the sample json to the DCL resource.

I wrote this by hand to get immediate functionality, but it should be generated by the template https://github.com/GoogleCloudPlatform/magic-modules/blob/master/tpgtools/templates/serialization.go.tmpl

There is a copy of the serialization code that was generated in tpgtools and then reimported to the DCL here: https://github.com/GoogleCloudPlatform/declarative-resource-client-library/blob/main/serialization/serialization.go#L36

The plan is to build our own copy with this JSON->DCL mapping, and regenerate it as new resources get added. Like how we need to generate code for DCL clients in config.go, but different in that we need to use it immediately for samples. So tpgtools serialization will need to run before tpgtools resource generation.